falcon/stream.py

```diff
@@ -45,21 +51,21 @@ class BoundedStream(io.IOBase):
 
     """
 
-    def __init__(self, stream, stream_len):
+    def __init__(self, stream: BinaryIO, stream_len: int) -> None:
         self.stream = stream
         self.stream_len = stream_len
 
         self._bytes_remaining = self.stream_len
 
-    def __iter__(self):
+    def __iter__(self) -> 'BoundedStream':
```

Comment:
it probably makes sense to import `from __future__ import annotated` so we can avoid the strings here in other places

Reply:
Done.